qcodes/dataset/descriptions.py

```diff
@@ -0,0 +1,67 @@
+import io
+from typing import Dict, Any
+
+from ruamel.yaml import YAML
+
+from qcodes.dataset.dependencies import InterDependencies
+
+
+class RunDescriber:
+    """
+    The object that holds the description of each run in the database. This
+    object serialises itself to a string and is found under the run_description
+    column in the runs table
+
+    Extension of this object is planned for the future, for now it holds the
+    parameter interdependencies
+    """
+
+    def __init__(self, interdeps: InterDependencies) -> None:
+        self.interdeps = interdeps
+
+    def serialize(self) -> Dict[str, Any]:
+        """
+        Serialize this object into a dictionary
+        """
+        ser = {}
+        ser['Parameters'] = self.interdeps.serialize()
+        return ser
+
+    @classmethod
+    def deserialize(cls, ser: Dict[str, Any]) -> 'RunDescriber':
+        """
+        Make a RunDescriber object based on a serialized version of it
+        """
+        idp = InterDependencies.deserialize(ser['Parameters'])
+        rundesc = cls(interdeps=idp)
+
+        return rundesc
+
+    def to_yaml(self):
+        """
+        Output the run description as a yaml string
+        """
+        yaml = YAML()
+        stream = io.StringIO()
```

Comment:
is it possible to do `with io.StringIO() as stream ... `?

Reply:
sure thing